### PR TITLE
Issue #665: Add automatic homebrew-core updates to CI

### LINKED ISSUE #665: Once git-machete is available in homebrew-core, replace the current code for updating the tap
- [x] Let's use `brew bump-formula-pr`, as suggested in https://github.com/Homebrew/homebrew-core/pull/111891#issuecomment-1261109350 instead. Worth checking what the linked Github Action does (even though we're not gonna use it, as we run on CircleCI).
- [x] README.md for installation via Homebrew
- [x] CONTRIBUTING.md about post-release steps
- [x] RELEASE_NOTES.md to mention the availability of the package via homebrew-core (as opposed to a tap)
- [x] Archive (**not** delete) [VirtusLab/homebrew-git-machete](https://github.com/VirtusLab/homebrew-git-machete) repo

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -156,7 +156,9 @@ Review fixes should be pushed on separate commits for easier viewing on GitHub (
    A bot is responsible for keeping the package up-to-date based on PyPI's release.
    Instruction on how to be a maintainer and other helpful tips can be found in
    [conda-forge recipe maintainer docs](https://conda-forge.org/docs/maintainer/adding_pkgs.html#recipe-maintainer).
-
+   
+1. Verify that there is an open PR in [homebrew-core](https://github.com/Homebrew/homebrew-core/pulls) titled: `git-machete <latest_version>`.
```
Comment (PawelLipski) [inline CONTRIBUTING.md]:
```suggestion
1. Verify that there is an open PR in [homebrew-core](https://github.com/Homebrew/homebrew-core/pulls?q=is%3Apr+git-machete) titled: `git-machete <latest_version>`.
```

File: README.md [hunk 2]
```diff
@@ -48,7 +48,6 @@ git-machete requires Python >= 3.6. Python 2.x is no longer supported.
 ### Using Homebrew (macOS)
 
 ```shell script
-brew tap VirtusLab/git-machete
```
Comment (PawelLipski) [inline README.md:51]: 🎉

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Release notes
 
+## New in git-machete 3.12.2
+
+- added: `git-machete` is now available in homebrew core formulae
```
Comment (PawelLipski) [inline RELEASE_NOTES.md]:
```suggestion
- added: `git-machete` is now available in homebrew core formulae; if `git-machete` has already been installed from the tap on a given machine, `brew` should automatically pull new updates from homebrew core from now on
```

File: ci/homebrew/ci-deploy.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+url="https://pypi.org/packages/source/g/git-machete/git-machete-$version.tar.gz"
+sha256=$(
+  curl -s https://$pypi_host/pypi/git-machete/$version/json \
+  | jq --raw-output '.urls | map(select(.packagetype == "sdist")) | .[0].digests.sha256')
+
+if [[ $do_push == true ]]; then
+  brew brew bump-formula-pr --url $url --sha256 $sha256 git-machete
+else
+  echo "Refraining from push since it's a dry run"
+  brew bump-formula-pr --dry-run --url $url --sha256 $sha256 git-machete
```
Comment (PawelLipski) [inline ci/homebrew/ci-deploy.sh:23]: Nice, what does this `--dry-run` do actually?

Comment (amalota) [reply ci/homebrew/ci-deploy.sh:23]: @PawelLipski ``` --dry-run       ->                 Print what would be done rather than doing it.```

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**